### PR TITLE
Simple example of memory free plugin

### PR DESCRIPTION
**Pull request by pitcons**

### CODE REVIEW

File: contrib/widget-examples/i3/mem-free.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
+mem_uev = io.open('/proc/meminfo', 'r')
+
+function get_mem_seg()
+    local result
```
Comment (shdown) [inline contrib/widget-examples/i3/mem-free.lua]: Please merge with the next line.

File: contrib/widget-examples/i3/mem-free.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
+mem_uev = io.open('/proc/meminfo', 'r')
```
Comment (shdown) [inline contrib/widget-examples/i3/mem-free.lua]: You should use `assert(io.open(…))` here, and not check `mem_uev` in `cb`.

File: contrib/widget-examples/i3/mem-free.lua [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        mem_uev:seek('set', 0)
+        for line in mem_uev:lines() do
+           local key, value, um = string.match(line, '(.*): +(.*) +(.*)')
+           if key == 'MemFree' then
+              value = tonumber(value)
+              if um == 'kB' then
+                 value = value / 1024.0 / 1024.0
+                 um = ' GB'
+              end
+              result = string.format('[%3.2f%s]', value, um)
```
Comment (shdown) [inline contrib/widget-examples/i3/mem-free.lua]: Probably you should add `break` here.

File: contrib/widget-examples/i3/mem-free.lua [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+function get_mem_seg()
+    local result = '[-?-]'
+    mem_uev:seek('set', 0)
+    for line in mem_uev:lines() do
+        local key, value, um = string.match(line, '(.*): +(.*) +(.*)')
+        if key == 'MemFree' then
+            value = tonumber(value)
+            if um == 'kB' then
+                value = value / 1024.0 / 1024.0
+                um = ' GB'
```
Comment (shdown) [inline contrib/widget-examples/i3/mem-free.lua]: It’s `'gB'`, not `' GB'`, I guess.

Comment (pitcons) [inline contrib/widget-examples/i3/mem-free.lua]: GB it's commonly used symbol (https://en.wikipedia.org/wiki/Gigabyte).

Comment (shdown) [inline contrib/widget-examples/i3/mem-free.lua]: Ok, but why the space?

Comment (pitcons) [inline contrib/widget-examples/i3/mem-free.lua]: Solely for reasons of good appearance :)

**Comment by shdown** [comment]
Merged, thanks.

**Comment by pitcons** [comment]
Thank you for you review!

**Pull request merged**